internal/vcs/metadata.go

```diff
@@ -223,12 +288,332 @@ func (f *metadataFetcher) getLocalGitMetadata(path string) (Metadata, error) {
+		logging.Logger.WithError(err).Debug("failed to ls remotes")
+	}
+
+	for _, rem := range rms {
+		urls := rem.Config().URLs
+		if len(urls) > 0 {
+			remote = urls[0]
+			break
+		}
+	}
+
 	return Metadata{
+		Remote: urlStringToRemote(remote),
 		Branch: Branch{Name: branch},
 		Commit: commitToMetadata(commit),
 	}, nil
 }
 
+func (f *metadataFetcher) getAzureReposMetadata(path string) (Metadata, error) {
+	m, err := f.getLocalGitMetadata(path)
+	if err != nil {
+		return m, fmt.Errorf("AzureRepos metadata error, could not fetch initial metadata from local git %w", err)
+	}
+
+	if m.Branch.Name == "HEAD" {
+		err := f.transformAzureDevOpsMergeCommit(path, &m)
+		if err != nil {
+			logging.Logger.WithError(err).Debug("could not transform Azure DevOps merge commit continuing with provided PR values")
+		}
+	}
+
+	res := f.getAzureRepoPRInfo()
+
+	m.Pipeline = &Pipeline{ID: os.Getenv("BUILD_BUILDID")}
+	pullID := os.Getenv("SYSTEM_PULLREQUEST_PULLREQUESTID")
+	m.PullRequest = &PullRequest{
+		ID:           pullID,
+		VCSProvider:  "azure_devops_tfsgit",
+		Title:        res.Title,
+		Author:       res.CreatedBy.UniqueName,
+		SourceBranch: strings.TrimLeft(os.Getenv("SYSTEM_PULLREQUEST_SOURCEBRANCH"), "refs/heads/"),
+		BaseBranch:   strings.TrimLeft(os.Getenv("SYSTEM_PULLREQUEST_TARGETBRANCH"), "refs/heads/"),
+		URL:          fmt.Sprintf("%s/pullrequest/%s", os.Getenv("SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI"), pullID),
+	}
+	return m, nil
+}
+
+type azurePullRequestResponse struct {
+	Repository struct {
+		Id      string `json:"id"`
+		Name    string `json:"name"`
+		Url     string `json:"url"`
+		Project struct {
+			Id          string `json:"id"`
+			Name        string `json:"name"`
+			Description string `json:"description"`
+			Url         string `json:"url"`
+			State       string `json:"state"`
+			Revision    int    `json:"revision"`
+		} `json:"project"`
+		RemoteUrl string `json:"remoteUrl"`
+	} `json:"repository"`
+	PullRequestId int    `json:"pullRequestId"`
+	CodeReviewId  int    `json:"codeReviewId"`
+	Status        string `json:"status"`
+	CreatedBy     struct {
+		Id          string `json:"id"`
+		DisplayName string `json:"displayName"`
+		UniqueName  string `json:"uniqueName"`
+		Url         string `json:"url"`
+		ImageUrl    string `json:"imageUrl"`
+	} `json:"createdBy"`
+	CreationDate  time.Time `json:"creationDate"`
+	Title         string    `json:"title"`
+	Description   string    `json:"description"`
+	SourceRefName string    `json:"sourceRefName"`
+	TargetRefName string    `json:"targetRefName"`
+	MergeStatus   string    `json:"mergeStatus"`
+	MergeId       string    `json:"mergeId"`
+}
+
+// getAzureRepoPRInfo attempts to get the azurePullRequestResponse using Azure DevOps Pipeline variables.
+// This method is expected to often fail as Azure DevOps requires users to explicitly pass System.AccessToken as
+// an env var on the job step.
+func (f *metadataFetcher) getAzureRepoPRInfo() azurePullRequestResponse {
```

Comment:
Just to keep naming consistent
```suggestion
func (f *metadataFetcher) getAzureReposPRInfo() azurePullRequestResponse {
```